daemon/checkpoint.go

```diff
@@ -8,6 +8,12 @@ import (
 	"path/filepath"
 
 	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/utils"
+)
+
+var (
+	validCheckpointNameChars   = utils.RestrictedNameChars
+	validCheckpointNamePattern = utils.RestrictedNamePattern
```

Comment:
Thanks @tiborvass @thaJeztah for the review. The PR has been updated.

I took a look at the leading `/`. It seems that container names are always stored in the name index with leading `/`. For example, `docker inspect my-container` will show
```
        "Name": "/my-container",
```
Though `docker ps` will only show `my-container` without leading `/`:
```
ubuntu@ubuntu:~$ docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS               NAMES
d9a020b06bbb        nginx:alpine        "nginx -g 'daemon off"   4 seconds ago       Up 3 seconds        80/tcp, 443/tcp     my-container
```

I will take a further look and see if there is any room for improvement.